namedb.h

```diff
@@ -150,6 +150,8 @@ struct zone
 	unsigned     is_skipped : 1; /* subsequent zone updates are skipped */
 	unsigned     is_checked : 1; /* zone already verified */
 	unsigned     is_bad : 1; /* zone failed verification */
+	char*        from_catalog; /* zone originates from catalog zone */
+	dname_type*  catalog_member_id; /* member id in the catalog zone */
```

Comment:
Is it wise to make `catalog_member_id` a pointer to the actual domain(?) We can then retrieve the `catalog` to which it belongs via that reference. Otherwise we'll probably need to do separate lookups every time, which can be costly(?) (Just a thought, don't know if it actually makes sense before I finish the review, might introduce a problem when removing zones...)